### PR TITLE
Improving log messages for missing summaries

### PR DESCRIPTION
**Pull request by wrwg**
## Description

This improves the reporting (currently via logs) of functions which have been flagged as angelic. Specifically, the type instantiation is printed. Example:

    function ShoppingCart::invariant can't be analyzed because it calls function std::ops::Deref::deref which has no body and no summary (foreign fn argument key: __alloc_vec_Vec_shopping_cart_Item)

Because the full type information seems to be erased from FunctionReference, we use the existing summary_argument_key. This is also what a user would have to use if he wants to create a foreign contract.

Also fixes a warning about overflow in the shopping cart example.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just keeping up with the latest Rustc nightly

## How Has This Been Tested?

./validate.sh and manual inspection of MIRAI run on shopping_cart example.

### CODE REVIEW

File: checker/src/visitors.rs [hunk 1]
```diff
@@ -3,6 +3,27 @@
 // This source code is licensed under the MIT license found in the
 // LICENSE file in the root directory of this source tree.
 
+use std::borrow::Borrow;
```
Comment (hermanventer) [inline checker/src/visitors.rs]: I like to keep these alphabetical and without random blank lines.

Comment (wrwg) [reply checker/src/visitors.rs]: I have not changed anything manually here. Some auto format in CLion did this (organize imports). The logic isn't random and the newlines have purpose (and are standard in many languages this way). Anyway, undone.

Comment (hermanventer) [reply checker/src/visitors.rs]: It seems like some great eminence decided that the std namespace needs special treatment. I can't say that I would like to adhere to this splendid wisdom.

Comment (wrwg) [reply checker/src/visitors.rs]: This goes back to java conventions: std libraries first, then 3rd party libraries, then your own project imports. It appears this has been taken over from CLion's roots (IntelliJ).

I love this feature. Automatic import management (adding, removing, ordering) is such a pleasure to work with in Java. I never look again at this part of the code (and it is actually folded away by default in intellij/android studio).

But when I have to look, if import lists become large, I also find it very useful for navigation.

Comment (hermanventer) [reply checker/src/visitors.rs]: Well, as long as Clippy and Format are happy, you are welcome put up a PR that sorts all of the import lists of all files in the repo.

**Pull request merged**